### PR TITLE
extracted ScanMatch method in LocalTrajectoryBuilder3D

### PR DESCRIPTION
**Pull request by schwoere**
Extracted the ScanMatch method in the LocalTrajectoryBuilder3D.

This is a short refactoring. With that change, the code of LocalTrajectoryBuilder3D resembles LocalTrajectoryBuilder2D more. (see LocalTrajectoryBuilder2D::ScanMatch)

### CODE REVIEW

File: cartographer/mapping/internal/3d/local_trajectory_builder_3d.cc [hunk 1]
```diff
@@ -54,6 +54,51 @@ LocalTrajectoryBuilder3D::LocalTrajectoryBuilder3D(
 
 LocalTrajectoryBuilder3D::~LocalTrajectoryBuilder3D() {}
 
+std::unique_ptr<transform::Rigid3d> LocalTrajectoryBuilder3D::ScanMatch(
+    const transform::Rigid3d& pose_prediction,
+    const sensor::PointCloud& low_resolution_point_cloud_in_tracking,
+    const sensor::PointCloud& high_resolution_point_cloud_in_tracking) {
+  if (low_resolution_point_cloud_in_tracking.empty()) {
```
Comment (wohe) [inline cartographer/mapping/internal/3d/local_trajectory_builder_3d.cc]: This is redundant. It is already verified before calling `ScanMatch()`. Remove similar to 2D?

**Comment by schwoere** [comment]
ptal @wohe

**Comment by schwoere** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user schwoere. Merge queue now has a length of 4.

**Comment by wally-the-cartographer** [comment]
Error: exit status 1

Log:
~~~
=> handle work item: PR 1245 on googlecartographer/cartographer.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at aa3ac7e8 Removed redundant 2d voxel filtering (#1243)
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Already on 'master'
Your branch is up to date with 'origin/master'.
 $ git pull [data/cartographer]
From https://github.com/googlecartographer/cartographer
   aa3ac7e8..3ee7c671  master     -> origin/master
Updating aa3ac7e8..3ee7c671
Fast-forward
 .../internal/3d/local_trajectory_builder_3d.cc     | 80 +++++++++++++++++++++-
 .../internal/3d/local_trajectory_builder_3d.h      |  5 +-
 2 files changed, 81 insertions(+), 4 deletions(-)
Current branch master is up to date.
 => Checking out schwoere/extract_scan_match_for_3d_traj_builder as local branch pr_1245.
 $ git remote add schwoere git@github.com:schwoere/cartographer.git [data/cartographer]
fatal: remote schwoere already exists.
 $ git fetch schwoere [data/cartographer]
 $ git branch --track pr_1245 schwoere/extract_scan_match_for_3d_traj_builder [data/cartographer]
Branch 'pr_1245' set up to track remote branch 'extract_scan_match_for_3d_traj_builder' from 'schwoere'.
 => Switching to branch pr_1245.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at 3ee7c671 Adding metrics  (#1240)
 $ git clean -fd [data/cartographer]
 $ git checkout pr_1245 [data/cartographer]
Switched to branch 'pr_1245'
Your branch is up to date with 'schwoere/extract_scan_match_for_3d_traj_builder'.
 $ git pull --force [data/cartographer]
Already up to date.
Current branch pr_1245 is up to date.
 => merge master.
 $ git merge master [data/cartographer]
Auto-merging cartographer/mapping/internal/3d/local_trajectory_builder_3d.h
Auto-merging cartographer/mapping/internal/3d/local_trajectory_builder_3d.cc
CONFLICT (content): Merge conflict in cartographer/mapping/internal/3d/local_trajectory_builder_3d.cc
Automatic merge failed; fix conflicts and then commit the result.
 => Abandoning branch pr_1245.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at 663582ae addressed comments
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
 $ git branch -D pr_1245 [data/cartographer]
Deleted branch pr_1245 (was 663582ae).

~~~

**Comment by schwoere** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user schwoere. Merge queue now has a length of 1.

**Pull request merged**